ktor-server/ktor-server-plugins/ktor-server-hsts/jvmAndNix/src/io/ktor/server/plugins/hsts/HSTS.kt

```diff
@@ -39,6 +40,18 @@ public class HSTSConfig {
      * Allows you to add custom directives supported by a specific user agent.
      */
     public val customDirectives: MutableMap<String, String?> = HashMap()
+
+    /**
+     * @see [withHost]
+     */
+    internal val hostSpecific: MutableMap<String, HSTSConfig> = HashMap()
+
+    /**
+     * Set specific configuration for a [host].
+     */
+    public fun withHost(host: String, configure: HSTSConfig.() -> Unit) {
```

Comment:
this allows writing code like 
```
install(HSTS) {
  withHost("a") { 
    withHost("b") { 
      ...
    }
  }
}
```
Can you fix this, please? Extracting base config should work.

Reply:
This has been changed and rebased.

Not sure what would be the most appropriate name for the base class, but though that `HSTSHostConfig` would be more indicative of function than something like`HSTSBaseConfig`